maker/src/trading/mod.rs

```diff
@@ -26,49 +31,54 @@ pub async fn run(orderbook_url: Url, maker_id: PublicKey, network: Network) -> R
 
     let orderbook_client = OrderbookClient::new();
 
-    let mut last_bid = None;
-    let mut last_ask = None;
-
-    while let Some(quote) = price_stream.try_next().await? {
-        tracing::debug!("Received new quote {quote:?}");
+    let mut orders: Vec<OrderResponse> = Vec::new();
 
-        last_bid = update_order(
+    // Closure to avoid repeating the same code
+    let add_new_order = |price, direction| {
+        add_order(
             &orderbook_client,
-            orderbook_url.clone(),
-            quote.bid(),
-            Direction::Long,
+            orderbook_url,
+            price,
+            direction,
             maker_id,
-            last_bid,
             dec!(1000),
         )
-        .await;
-        last_ask = update_order(
-            &orderbook_client,
-            orderbook_url.clone(),
-            quote.ask(),
-            Direction::Short,
-            maker_id,
-            last_ask,
-            dec!(1000),
-        )
-        .await;
+    };
+
+    while let Some(quote) = price_stream.try_next().await? {
```

Comment:
🤔 Beforehand we were firstly adding the new order and then deleting the old ones. Not sure if that was intentional, but this change is reversing the order now.

Reply:
yes, it is intentional - we should first remove the outdated order (we don't want anyone to take it, because the price has changed in the event). 

I will mention this in the commit message to be clear about it.

Reply:
Could we split the change into two then? Otherwise it looks like the change is purely a refactor if you look at the commit title.

Reply:
I don't think it's worth it, but I'll try to be more atomic in the future!

Reply:
At least let's rename the commit so that we can more easily spot the change in the future, since we know it's not a refactor.

Reply:
suggestions welcome!

Reply:
`fix(maker): Delete stale orders before sending new ones`.